tests/target/big-impl-block.rs

```diff
@@ -63,11 +60,8 @@ impl Foo<ExcessivelyLongGenericName, ExcessivelyLongGenericName, AnotherExcessiv
     > {
     fn foo() {}
 }
-impl<
-    ExcessivelyLongGenericName,
-    ExcessivelyLongGenericName,
-    AnotherExcessivelyLongGenericName,
-> Foo<ExcessivelyLongGenericName, ExcessivelyLongGenericName, AnotherExcessivelyLongGenericName>
+impl<ExcessivelyLongGenericName, ExcessivelyLongGenericName, AnotherExcessivelyLongGenericName>
+    Foo<ExcessivelyLongGenericName, ExcessivelyLongGenericName, AnotherExcessivelyLongGenericName>
     for Bar<
         ExcessivelyLongGenericName,
         ExcessivelyLongGenericName,
```

Comment:
I realise that these are technically different kinds of type parameter, however, it seems strange that they are formatted differently when they look so similar. I think by default we should format type params one per line if they don't fit on a single line.

Reply:
Thank you for your review!

Would you please clarify which part you think is strange? Both `imp<_>` and `Foo<_>` are formatted on a single line, so I think they are ok. Do you mean that we should treat `impl<_> Foo<_>` as a whole and format them like a function signature?

Reply:
By function signature, we use multiple lines for arguments when the whole signature does not fit in a single line. Do you think that we should appli similar rule and break the `impl<..>` first? For example, one of the followings:

```rust
// Break `impl<..>`, keep `Foo<..>` in a single line as long as it fits within the max width.
impl<
    ExcessivelyLongGenericName,
    ExcessivelyLongGenericName,
    AnotherExcessivelyLongGenericName,
> Foo<ExcessivelyLongGenericName, ExcessivelyLongGenericName, AnotherExcessivelyLongGenericName>

// Break both `impl<..>` and `Foo<..>` if they do not fit within the max width.
impl<
    ExcessivelyLongGenericName,
    ExcessivelyLongGenericName,
    AnotherExcessivelyLongGenericName,
> Foo<
    ExcessivelyLongGenericName,
    ExcessivelyLongGenericName,
    AnotherExcessivelyLongGenericName,
> 
```

We use the former rule for function signatur.

Reply:
"Would you please clarify which part you think is strange?" - that `Bar<...>` is formatted differently to `Foo<...>` (and I was wrong - they are the same kind of type params).

I think we should probably break all param lists if we break any of them at least if they are similar? Perhaps if we put the types on different lines then we should format using the space available in the shortest line (in this example, that would be the space following `for`, I think).

If this looks hard, or you don't think it is a problem, then I don't think it matters too much, I don't think we're likely to hit this case in real life.

Reply:
Thank you for clarifying. I don't think this is a problem since this seems super rare and one should use `where` for this kind of cases.